src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.ComposeCache.targets

```diff
@@ -0,0 +1,370 @@
+          UseSymboliclinksIfPossible="$(CreateSymbolicLinksForPublishFilesIfPossible)">
+
+      <Output TaskParameter="DestinationFiles" ItemName="FileWrites"/>
+
+    </Copy>
+
+  </Target>
+
+  <!--
+    ============================================================
+                                        _ComputeResolvedFilesToCachTypes
+
+    Splits ResolvedFileToPublish items into 'managed' and 'unmanaged' buckets.
+    ============================================================
+    -->
+  <Target Name="_ComputeResolvedFilesToCachTypes">
+    <ItemGroup>
+      <_ManagedResolvedFileToPublishCandidates Include="@(ResolvedFileToPublish)"
+                                             Condition="'%(ResolvedFileToPublish.AssetType)'=='runtime'" />
+
+      <_UnManagedResolvedFileToPublish Include="@(ResolvedFileToPublish)"
+                                     Condition="'%(ResolvedFileToPublish.AssetType)'!='runtime'" />
+    </ItemGroup>
+  </Target>
+
+
+  <!--
+    ============================================================
+                                        ComputeFilesToCache
+
+    Gathers all the files that need to be copied to the publish directory.
+    ============================================================
+    -->
+  <Target Name="ComputeFilesToCache"
+          DependsOnTargets="_ComputeNetPublishAssets;
+                            _ComputeCopyToPublishDirectoryItems">
+
+    <PropertyGroup>
+      <CopyBuildOutputToPublishDirectory Condition="'$(CopyBuildOutputToPublishDirectory)'==''">true</CopyBuildOutputToPublishDirectory>
+      <CopyOutputSymbolsToPublishDirectory Condition="'$(CopyOutputSymbolsToPublishDirectory)'==''">true</CopyOutputSymbolsToPublishDirectory>
+    </PropertyGroup>
+
+    <ItemGroup>
+
+      <!-- Copy all the assemblies -->
+      <ResolvedFileToPublish Include="@(ResolvedAssembliesToPublish)">
+      </ResolvedFileToPublish>
+    </ItemGroup>
+
+  </Target>
+
+  <!--
+    ============================================================
+                                        _RunOptimizer
+
+    Start the optimization face
+    ============================================================
+    -->
+  <Target Name="_RunOptimizer"
+          DependsOnTargets="_ComputeResolvedFilesToCachTypes;
+                            _GenerateCrossgenProj;
+                            _SetupCrossgenStage;
+                            _RestoreCrossgen">
+    <!-- Get the coreclr path -->
+    <ItemGroup>
+      <_CoreclrResolvedPath Include="@(CrossgenResolvedAssembliesToPublish)"
+                                             Condition="'%(CrossgenResolvedAssembliesToPublish.Filename)'=='coreclr'" />
+      <_CoreclrResolvedPath Include="@(CrossgenResolvedAssembliesToPublish)"
+                                             Condition="'%(CrossgenResolvedAssembliesToPublish.Filename)'=='libcoreclr'" />
+      <_AssembliestoCrossGen Include="$(_RuntimeDir)\**\*.*" />
```

Comment:
Only the managed assemblies, which are copied in step https://github.com/dotnet/sdk/pull/591/files/f8e0bd7d906be1ae3d1f8b105bf539455c3da6ce#diff-8a3d011306d4fb49ae22e3b9f51e2811R309